File: C++/2-Medium/00165._比较版本号/00165._比较版本号.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

// 分割字符串函数，以 '.' 为分隔符
vector<int> splitVersion(const string& version) {
    vector<int> nums;
    int start = 0;
    for(int i=0; i <= version.size(); ++i){
        if(i==version.size() || version[i] == '.'){
            nums.push_back(stoi(version.substr(start, i-start)));
            start = i + 1;
        }
    }
    return nums;
}

// 比较版本号
int compareVersion(string version1, string version2) {
    vector<int> nums1 = splitVersion(version1), nums2 = splitVersion(version2);
    int n1 = nums1.size(), n2 = nums2.size();
    
    // 比较每个修订号
    for (int i = 0; i < max(n1, n2); ++i) {
        // 获取当前位置的修订号，如果超出版本号长度，则为0
        int v1 = i < n1 ? nums1[i] : 0;
        int v2 = i < n2 ? nums2[i] : 0;
        if (v1 < v2) return -1;
        else if (v1 > v2) return 1;
    }
    return 0; // 所有修订号都相等
}

int main() {
    // 示例测试
    cout << compareVersion("1.01", "1.001") << endl; // 应输出 0
    cout << compareVersion("1.0", "1.0.0") << endl;  // 应输出 0
    cout << compareVersion("0.1", "1.1") << endl;    // 应输出 -1
    return 0;
}

File: Python/2-Medium/00165._比较版本号/00165._比较版本号.py
from typing import List

def split_and_normalize(version: str) -> List[int]:
    """
    分割并标准化版本号字符串。
    
    Args:
        version: 一个版本号字符串。
        
    Returns:
        一个整数列表，表示标准化后的版本号各段。
    
    说明:
        - 分割字符串，使用'.'作为分隔符。
        - 转换每段为整数，以去除前导零。
    """
    return [int(v) for v in version.split('.')]

def compare_version(version1: str, version2: str) -> int:
    """
    比较两个版本号。
    
    Args:
        version1: 第一个版本号字符串。
        version2: 第二个版本号字符串。
        
    Returns:
        - 如果version1 > version2，返回1。
        - 如果version1 < version2，返回-1。
        - 否则，返回0。
        
    说明:
        - 首先分割并标准化两个版本号。
        - 然后逐段比较，直到找到差异或遍历完所有段。
    """
    v1 = split_and_normalize(version1)
    v2 = split_and_normalize(version2)
    len_v1, len_v2 = len(v1), len(v2)
    
    for i in range(max(len_v1, len_v2)):
        num1 = v1[i] if i < len_v1 else 0
        num2 = v2[i] if i < len_v2 else 0
        
        if num1 > num2:
            return 1
        elif num1 < num2:
            return -1
            
    return 0  # 所有段都相等

# 示例运行
results = {
    "1.01 vs 1.001": compare_version("1.01", "1.001"),  # 应该输出: 0
    "1.0 vs 1.0.0": compare_version("1.0", "1.0.0"),    # 应该输出: 0
    "0.1 vs 1.1": compare_version("0.1", "1.1")         # 应该输出: -1
}
print(results)